### DISCUSSION TITLE
Black window in Steam

### DISCUSSION THREAD

**Discussion by mohd-akram** [The-Wineskin-Project/WineskinServer#344] [Q&A] [answered]
I created a new wrapper (WS11WineCX64Bit22.0.1-1) and installed Steam via winetricks. I then added `-udpforce -allosarches -cef-force-32bit` as mentioned in the wiki. I still had to add `-noreactlogin` to not get a black login window. After logging in, the Store, Library, etc. tabs just show a black pane. This used to work at some point, I'm not sure what's changed. I'm on macOS 12.6.2 (Intel).

**Comment by asmaloney** [comment]
I've had the same problem and have not found a solution. Works sometimes, not others. Unfortunately my experience of the past year or so is that Steam and Epic Game Store (even though you didn't mention it 😆) are too flaky to rely on (on my machine at least - macOS 12.6.2 Intel as well).

**Reply by Gcenx** [reply]
Epic Games Launcher isn’t supported even in CrossOver, there’s opted to instead make use of the Heroic Games launcher.

When I’ve got enough time to complete the next wrapper update it should be possible to use the latest version of Heroic Games Launcher directly on Wineskin wrappers.

**Reply by asmaloney** [reply]
Oh cool! Thanks Dean.

I tried Heroic Games launcher a while ago but couldn't get it to recognize my installed games (and couldn't get any answers), so I moved on to other things...

I'll give it a try again sometime.

**Comment by Gcenx** [comment]
Weirdly I don’t have any issues with Steam, not with new wrapper nor even my old one that I keep updating.

CX22.1.0 seems to be forcing the “new big picture mode” (DeckUI) and that’s working but that requires the Steam beta client along with an additional command line option `-newbigpicture`

**Reply by Gcenx** [reply]
Instead of an edit replying to my own comment.

The annoyance of the current beta client is the removal of `-noreactlogin`

**Comment by asmaloney** [comment]
I had some time to try again. I started from scratch with:

- macOS: 12.6.3 (Intel)
- Wineskin: 2.9.1.5
- Engine: WS11WineCX64Bit22.0.1-1

I followed the instructions [from here](https://github.com/Gcenx/WineskinServer/wiki):
- updated Winetricks
- installed `dxvk11-3_macOS` 
- installed `Steam`
- browsed for app and selected `"C:\Program Files (x86)\Steam\Steam.exe"`
- added `-udpforce -allosarches -cef-force-32bit` to this

Running it the first time, the login screen was black. I added `-noreactlogin` and restart and it let me log in.

Interface came up but the content part (where the library should be) was solid black.

Went to the settings and turned off `Enable GPU accelerated rendering in web views` & restarted.

Now the content area is solid white.

So I still can't get it to work with the latest versions & instructions 😢

**Reply by Gcenx** [reply]
I’d tested Steam last night and here’s my steps using `WS11WineCX64Bit22.0.1-1`

This test was on an M1 mini running macOS Ventura 13.2.

Create a blank wrapper, open winetricks and install the following verbs
- steam
- vcrun2019
- dsetup
- dxvk1103_macOS

Manually select steam.exe and add the launch flags, launch and allow Steam to get the latest version then exit.

Relaunch and login using the QR code.

Go to settings and turned off `Enable GPU accelerated rendering in web views` & restart steam.

After this Steam Store/Library etc was fully functional.

**Reply by asmaloney** [reply]
Thanks Dean. That worked! (None of my Steam games work, but 🤷 😆 )

The differences from when I tried are:

- now using `Wineskin-2.9.1.6`
- now running `vcrun2019` and `dsetup` verbs
- you updated DXVK in `dxvk1103_macOS` a couple of days ago

(Wineskin also crashed when I clicked "Test Run" but it continued the download.)

**Reply by Gcenx** [reply]
> Wineskin also crashed when I clicked "Test Run" but it continued the download.)

For something like Steam it’s best to avoid “Test Run”, there’s a checkbox to always save logs you’d want to check that and then run normally and it will save logs like “Test Run” without it deciding to die over size of the logs

**Reply by Gcenx** [reply]
> you updated DXVK in dxvk1103_macOS a couple of days ago

Yeah it got broken during the last rebase so since I was fixing it anyway thought I’d swap to the latest archive in preparation for CX22.1.0

**Reply by mohd-akram** [reply]
The steps worked for me as well, thank you very much. I can't seem to mark your reply as an answer though, perhaps post it in a separate comment. Thank you @asmaloney for investigating as well.

**Comment by mohd-akram** [comment, marked as answer]
[Per](https://github.com/Gcenx/WineskinServer/discussions/344#discussioncomment-4831972) @Gcenx:

> Create a blank wrapper, open winetricks and install the following verbs
> 
> * steam
> * vcrun2019
> * dsetup
> * dxvk1103_macOS
> 
> Manually select steam.exe and add the launch flags, launch and allow Steam to get the latest version then exit.
> 
> Relaunch and login using the QR code.
> 
> Go to settings and turned off `Enable GPU accelerated rendering in web views` & restart steam.

**Reply by Gcenx** [reply]
As long as your running macOS Catalina and later the newly pushed `WS11WineCX64Bit22.0.1-2` engine shouldn’t require additional steps

Simply installing the `steam` verb and adding the launch commands should now be enough, at least it was from my testing last night.

**Comment by OrandelP** [comment]
I'm stuck here. I've followed all of the steps, but after I click the main wrapper, a stream box appears that looks like this:

https://imgur.com/ZOSDTL2

No QR Code to scan.

I've tried adding

steam

vcrun2019

dsetup &

dxvk1103_macOS

But Still the black screen (also I can't see any text).

https://imgur.com/a/82BMQph

Am I adding these correctly? (I even added -noreactlogin)

https://imgur.com/a/NOCgZqu

**Reply by Gcenx** [reply]
These images are not very clear due to the image host.

**Comment by mohd-akram** [comment]
It seems there's no need for `-allosarches -cef-force-32bit` anymore (the CEF bug has been fixed AFAIK)  and it runs fine without them for me yet Crossover Wine [still hardcodes them](https://github.com/Gcenx/winecx/blob/crossover-wine-24.0.3/dlls/kernelbase/process.c#L763-L789) when it detects `steam.exe` being run. I had to `ln -s steam.exe vapor.exe` and run that to avoid them being added. The 64-bit version seems to run better for me.

@Gcenx Perhaps that hack should be removed in the Wineskin version.

**Reply by Gcenx** [reply]
I’m aware, removing this would require me to bump the current Engines.
